sympy/tensor/array/array_comprehension.py

```diff
@@ -272,4 +273,59 @@ def _array_subs(arr, var, val):
                     list_gen.append(list_expr.subs(var, val))
                 else:
                     list_gen.append(_array_subs(list_expr, var, val))
-        return ImmutableDenseNDimArray(list_gen)
+        return list_gen
```

Comment:
Why `ImmutableDenseNDimArray` is shifted to `doit()`? Something related to performance?

Reply:
Ah! got it, it is being used in `tomatrix`, etc. So returning a generalized expression for type casting. No issues. Resolve this conversation.